src/Autofac/Features/OpenGenerics/OpenGenericServiceBinder.cs

```diff
@@ -173,7 +173,18 @@ internal static class OpenGenericServiceBinder
 
         if (!serviceType.IsInterface)
         {
-            return TryFindServiceArgumentsForImplementation(implementationType, serviceGenericArguments, serviceTypeDefinition.GetGenericArguments());
+            var baseType = GetGenericBaseType(implementationType, serviceTypeDefinition);
```

Comment:
Can I request a comment here explaining 'why' we have to do this? This generic stuff is already pretty complex, and without scouting through the tests/issues, not sure it will be clear (perhaps just tag the issue number in a comment)?

Reply:
Totally fair and excellent point. I'll get something in there.

Reply:
Minor clarifying manifesto added to explain what's going on here and why.

Reply:
Thanks, very useful comment.